pkg/plugins/scms/github/pullrequest.go

```diff
@@ -265,13 +296,30 @@ func (p *PullRequest) updatePullRequest() error {
 // EnablePullRequestAutoMerge updates an existing pullrequest with the flag automerge
 func (p *PullRequest) EnablePullRequestAutoMerge() error {
 
+	// Test that automerge feature is enabled on repository but only if we plan to use it
+	autoMergeAllowed, err := p.isAutoMergedEnabledOnRepository()
+	if err != nil {
+		return err
+	}
+
+	if !autoMergeAllowed {
+		return ErrAutomergeNotAllowOnRepository
+	}
+
 	var mutation mutationEnablePullRequestAutoMerge
 
 	input := githubv4.EnablePullRequestAutoMergeInput{
 		PullRequestID: githubv4.String(p.remotePullRequest.ID),
 	}
 
-	err := p.gh.client.Mutate(context.Background(), &mutation, input, nil)
+	// The Github Api expects the merge method to be capital letter and don't allows empty value
+	// hence the reason to set input.MergeMethod only if the value is not nil
+	if len(p.spec.MergeMethod) > 0 {
```

Comment:
```suggestion
	if p.spec.MergeMethod != "" {
```

NITPICK: "Idiomatic" way to check for empty strings: when reading it clear that `p.spec.MergeMethod` is a string (while `len(p.spec.MergeMethod)` is ambiguous: is it a slice or a string or something else ?)